src/input/agent/cost_limit.rs

```diff
@@ -0,0 +1,199 @@
+//! Code for reading the agent cost limits CSV file.
+use super::super::*;
+use crate::agent::{AgentCostLimits, AgentCostLimitsMap, AgentID};
+use crate::id::IDCollection;
+use crate::year::parse_year_str;
+use anyhow::{Context, Result};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::path::Path;
+
+const AGENT_COST_LIMITS_FILE_NAME: &str = "agent_cost_limits.csv";
+
+#[derive(PartialEq, Debug, Deserialize)]
+struct AgentCostLimitsRaw {
+    agent_id: String,
+    year: String,
+    capex_limit: Option<f64>,
+    annual_cost_limit: Option<f64>,
+}
+
+impl AgentCostLimitsRaw {
+    fn to_agent_cost_limits(&self) -> AgentCostLimits {
+        AgentCostLimits {
+            capex_limit: self.capex_limit,
+            annual_cost_limit: self.annual_cost_limit,
+        }
+    }
+}
+
+/// Read agent cost limits info from the agent_cost_limits.csv file.
+///
+/// # Arguments
+///
+/// * `model_dir` - Folder containing model configuration files
+///
+/// # Returns
+///
+/// A map of Agents, with the agent ID as the key
+pub fn read_agent_cost_limits(
+    model_dir: &Path,
+    agent_ids: &HashSet<AgentID>,
+    milestone_years: &[u32],
+) -> Result<HashMap<AgentID, AgentCostLimitsMap>> {
+    let file_path = model_dir.join(AGENT_COST_LIMITS_FILE_NAME);
+    let agent_cost_limits_csv = read_csv(&file_path)?;
```

Comment:
So it *looks* like we allow for users *not* supplying cost limits for a particular agent (in which case they get the defaults), but if the CSV file is totally empty then users will get an error.

Would it be better to do:
```suggestion
    let agent_cost_limits_csv = read_csv_optional(&file_path)?;
```